src/main/java/sirius/kernel/nls/Formatter.java

```diff
@@ -211,6 +227,10 @@ public Formatter setDirect(Map<String, Object> ctx) {
      * @return <tt>this</tt> to permit fluent method chains
      */
     public Formatter setDirect(String property, String value) {
+        if (jsEncode && value != null) {
+            replacement.put(property, value.replace("'", "\\'").replace("\"", "\\\""));
+            return this;
+        }
         replacement.put(property, urlEncode ? Strings.urlEncode(value) : value);
```

Comment:
shouldn't we also make the if urlEncode part more explicit? like

if (... != null) {

 if (url) { ...}
if (js) {...}
}

...